### DISCUSSION TITLE
Can't get the web-api to work

### DISCUSSION THREAD

**Discussion by duncrow** [cea-sec/ivre#1497] [Q&A] [answered]
Hi all,

I feel quite stupid, but I don't get results via the web-api. After reading the docs, I think the following should work (But, as you see, I just get a 404 error. Working against localhost to avoid reverse-proxy issues.)

```
$ curl -X GET http://localhost/config
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>404 Not Found</title>
</head><body>
<h1>Not Found</h1>
<p>The requested URL was not found on this server.</p>
</body></html>
```

All other api calls documented here https://doc.ivre.rocks/en/latest/dev/web-api.html don't work either (404). The normal web-frontend works just fine otherwise, no problems there. What am I missing?

General info (system is running fine with mongo-db, fully patched Debian 10.13):
```
$ ivre version
IVRE - Network recon framework
Copyright 2011 - 2023 Pierre LALET <pierre@droids-corp.org>
Version 0.9.20.dev32

Python 3.7.3 (default, Oct 31 2022, 14:04:00)
[GCC 8.3.0]

Linux IVRE 4.19.0-23-amd64 #1 SMP Debian 4.19.269-1 (2022-12-20) x86_64

Dependencies:
    MySQLdb: *missing*
    OpenSSL: 23.1.1
    PIL: *missing*
    bottle: 0.12.25
    cryptography: 40.0.1
    dbus: 1.2.8
    krbV: *missing*
    matplotlib: *missing*
    psycopg2: *missing*
    pycurl: PycURL/7.43.0.2 libcurl/7.64.0 GnuTLS/3.6.7 zlib/1.2.11 libidn2/2.0.5 libpsl/0.20.2 (+libidn2/2.0.5) libssh2/1.8.0 nghttp2/1.36.0 librtmp/2.3
    pymongo: 4.3.3
    sqlalchemy: *missing*
    tinydb: *missing*
```

**Comment by p-l-** [comment, marked as answer]
Hey there!

I can see two problems here:
- the API is usually mounted under `/cgi`;
- you usually need a specific `Referer` header value for the anti-CSRF check.

You should try something like:
```
curl -v http://localhost/cgi/view/count -H 'Referer: http://localhost/'
```
